packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -110,15 +99,38 @@ describe('<DataGrid /> - Pagination', () => {
       expect(onPageChange.callCount).to.equal(2);
     });
 
-    it('should not trigger onPageChange on initialisation and rendering of the first and default page', () => {
+    it('should not trigger onPageChange on initialisation or on page prop change', () => {
       const onPageChange = spy();
 
-      render(
-        <div style={{ width: 300, height: 300 }}>
-          <DataGrid {...baselineProps} onPageChange={onPageChange} pageSize={1} />
-        </div>,
-      );
+      function Test({page, pageSize}) {
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <DataGrid {...baselineProps} onPageChange={onPageChange} pageSize={pageSize} page={page} />
```

Comment:
It could be simpler to spread all the props

```suggestion
            <DataGrid {...baselineProps} onPageChange={onPageChange} {...props} />
```

x2